src/main/kotlin/com/sourcegraph/cody/chat/actions/BaseCommandAction.kt

```diff
@@ -1,17 +1,34 @@
 package com.sourcegraph.cody.chat.actions
 
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.fileEditor.FileDocumentManager
 import com.intellij.openapi.fileEditor.FileEditorManager
 import com.intellij.openapi.project.Project
 import com.sourcegraph.cody.CodyToolWindowContent
+import com.sourcegraph.cody.agent.protocol.ProtocolTextDocument
 import com.sourcegraph.cody.chat.AgentChatSession
 import com.sourcegraph.cody.commands.CommandId
+import com.sourcegraph.cody.context.ui.ActionInIgnoredFileNotification
+import com.sourcegraph.cody.ignore.IgnoreOracle
+import com.sourcegraph.cody.ignore.IgnorePolicy
 
 abstract class BaseCommandAction : BaseChatAction() {
 
   abstract val myCommandId: CommandId
 
   override fun doAction(project: Project) {
-    FileEditorManager.getInstance(project).selectedTextEditor?.let {
+    FileEditorManager.getInstance(project).selectedTextEditor?.let { editor ->
+      ApplicationManager.getApplication().assertIsDispatchThread()
```

Comment:
Actually `selectedTextEditor` should be called on EDT only as well, so you could move that assertion to the first line of the method.